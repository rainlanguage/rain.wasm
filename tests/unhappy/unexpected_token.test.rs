#[macro_use]
extern crate wasm_bindgen_utils;

struct SomeType;
impl_wasm_traits!(SomeType,);
impl_wasm_traits!(SomeType.);
impl_wasm_traits!(SomeType/);
impl_wasm_traits!(SomeType;);
impl_wasm_traits!(SomeType:);
impl_wasm_traits!(SomeType::);
impl_wasm_traits!(SomeType());
impl_wasm_traits!(SomeType<>);

fn main() {}
