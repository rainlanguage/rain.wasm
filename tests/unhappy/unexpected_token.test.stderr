error: no rules expected the token `,`
 --> tests/unhappy/unexpected_token.test.rs:5:27
  |
5 | impl_wasm_traits!(SomeType,);
  |                           ^ no rules expected this token in macro call
  |
  = note: while trying to match sequence start

error: no rules expected the token `.`
 --> tests/unhappy/unexpected_token.test.rs:6:27
  |
6 | impl_wasm_traits!(SomeType.);
  |                           ^ no rules expected this token in macro call
  |
  = note: while trying to match sequence start

error: no rules expected the token `/`
 --> tests/unhappy/unexpected_token.test.rs:7:27
  |
7 | impl_wasm_traits!(SomeType/);
  |                           ^ no rules expected this token in macro call
  |
  = note: while trying to match sequence start

error: no rules expected the token `;`
 --> tests/unhappy/unexpected_token.test.rs:8:27
  |
8 | impl_wasm_traits!(SomeType;);
  |                           ^ no rules expected this token in macro call
  |
  = note: while trying to match sequence start

error: no rules expected the token `:`
 --> tests/unhappy/unexpected_token.test.rs:9:27
  |
9 | impl_wasm_traits!(SomeType:);
  |                           ^ no rules expected this token in macro call
  |
  = note: while trying to match sequence start

error: no rules expected the token `::`
  --> tests/unhappy/unexpected_token.test.rs:10:27
   |
10 | impl_wasm_traits!(SomeType::);
   |                           ^^ no rules expected this token in macro call
   |
   = note: while trying to match sequence start

error: no rules expected the token `(`
  --> tests/unhappy/unexpected_token.test.rs:11:27
   |
11 | impl_wasm_traits!(SomeType());
   |                           ^ no rules expected this token in macro call
   |
   = note: while trying to match sequence start

error: no rules expected the token `>`
  --> tests/unhappy/unexpected_token.test.rs:12:28
   |
12 | impl_wasm_traits!(SomeType<>);
   |                            ^ no rules expected this token in macro call
   |
note: while trying to match meta-variable `$generics:ident`
  --> src/macros.rs
   |
   |     ($type_name:ident $(< $($generics:ident),+ >)?) => {
   |                             ^^^^^^^^^^^^^^^
